websites/C/Coursera/presence.ts

```diff
@@ -0,0 +1,341 @@
+type Functionlize<T> = {
+	[P in keyof T]: () => T[P];
+};
+
+interface Route extends Functionlize<Partial<PresenceData>> {
+	path: RegExp;
+	playback?(): boolean;
+	run?(): PresenceData;
+}
+
+const enum Assets {
+	Logo = "https://i.imgur.com/P4XquqE.png",
+}
+
+enum Settings {
+	TIMESTAMP = "timestamp",
+	BUTTONS = "buttons",
+}
+const { href } = document.location,
+	presence = new Presence({
+		clientId: "1179996601327026227",
+	}),
+	startTimestamp: number = Math.floor(Date.now() / 1000),
+	router = ({ path }: { path: string; presenceData: PresenceData }): Route => {
+		const routes: Route[] = [
+			{
+				path: /^\/$/,
+				details: () => "On Homepage",
+				smallImageKey: () => Assets.Reading,
+				smallImageText: () => "Browsing",
+			},
+
+			{
+				path: /^\/account-profile$/,
+				details: () => " Viewing own Profile",
+				smallImageKey: () => Assets.Reading,
+				smallImageText: () => "Browsing",
+			},
+			{
+				path: /^\/user\/.*$/,
+				details: () => {
+					return `Viewing ${
+						document.querySelector(
+							"[data-testid='profile-photo-invite-section-title']"
+						).textContent
+					}'s Profile`;
+				},
+				smallImageKey: () => Assets.Reading,
+				smallImageText: () => "Browsing",
+				buttons: () => [{ label: "Profile Link", url: href }],
```

Comment:
Button labels should be an action.
```suggestion
				buttons: () => [{ label: "View Profile", url: href }],
```